src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = new Node[INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndex(key);
+        if (size == (capacity * LOAD_FACTOR)) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> nodeByIndex = table[index];
+            while (nodeByIndex != null) {
+                if (Objects.equals(nodeByIndex.key, key)) {
+                    nodeByIndex.value = value;
+                    return;
+                }
+                if (nodeByIndex.next == null) {
+                    nodeByIndex.next = newNode;
+                    size++;
+                    return;
+                }
+                nodeByIndex = nodeByIndex.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        capacity = capacity * RESIZE_FACTOR;
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
```

Comment:
after removing field `capacity` u can write 
```suggestion
        Node<K, V>[] oldTable = table;
        table = new Node[table.length * 2];
```